src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -0,0 +1,390 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.repair.autorepair;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.exceptions.ConfigurationException;
+
+public class AutoRepairConfig implements Serializable
+{
+    // enable/disable auto repair globally, overrides all other settings. Cannot be modified dynamically.
+    public final Boolean enabled;
+    // the interval in seconds between checks for eligible repair operations. Cannot be modified dynamically.
+    public final Integer repair_check_interval_in_sec = 300; // 5 minutes
```

Comment:
this is kinda across all of these, but if look at Config.java can see now that we for times instead of having _in_sec just pass a duration type. Should probably stick with this kinda thing for most of the config options

Reply:
Updated the configs to match with the 4.x by using the `DurationSpec.*`